kyaml/yaml/fns.go

```diff
@@ -725,7 +725,7 @@ func (s FieldSetter) Filter(rn *RNode) (*RNode, error) {
 	}
 
 	// Clear the field if it is empty, or explicitly null
-	if s.Value == nil || s.Value.IsTaggedNull() {
```

Comment:
There are three possible ways to set s.Value.

1. Set the `Value` field directly
2. Use [the SetField function](https://github.com/kubernetes-sigs/kustomize/blob/779f1530719c61df642e69675c7c47c87e4a1267/kyaml/yaml/fns.go#L613-L615)
3. Use [the Set function](https://github.com/kubernetes-sigs/kustomize/blob/779f1530719c61df642e69675c7c47c87e4a1267/kyaml/yaml/fns.go#L617-L619)

I've checked every case where s.Value could be null and found in two cases.

1. Walker#walkMap